pkg/controllers/clusterresourceplacement/resource_selector.go

```diff
@@ -357,6 +357,8 @@ func (r *Reconciler) shouldSelectResource(gvr schema.GroupVersionResource) bool
 
 // generateRawContent strips all the unnecessary fields to prepare the objects for dispatch.
 func generateRawContent(object *unstructured.Unstructured) ([]byte, error) {
+	// Make a deep copy of the object as we are modifying it.
+	object = object.DeepCopy()
```

Comment:
who else is read/write it?

Reply:
The informer. I put the story in the PR description.

Reply:
the go-client informer returns a deep copy of the object so there is no race condition between that.

Reply:
The informer lister lists resources from the indexer, which is backed by the underlying threadsafestore. Reading the doc: https://pkg.go.dev/k8s.io/client-go/tools/cache#ThreadSafeStore, "TL;DR caveats: you must not modify anything returned by Get or List as it will break the indexing feature in addition to not being thread safe." It's required to keep the objects r/o. I don't see a deep copy as I dig the code path.